components/builder-api-client/src/builder.rs

```diff
@@ -1142,4 +1175,100 @@ mod tests {
+                               .collect::<Vec<_>>();
+
+            if filtered.is_empty() {
+                return Ok((PackageResults { range_start: 0,
+                                            range_end:   0,
+                                            total_count: 0,
+                                            data:        vec![], },
+                           false));
+            }
+
+            let total = filtered.len();
+            let last = total - 1;
+            let (start, end) = if range >= last {
+                (last, last)
+            } else {
+                (range, (range + step).min(last))
+            };
+            let filtered_range = filtered[start..=end].iter()
+                                                      .map(|s| get_test_ident(**s))
+                                                      .collect::<Vec<_>>();
+            let result = PackageResults { range_start: start as isize,
+                                          range_end:   end as isize,
+                                          total_count: total as isize,
+                                          data:        filtered_range, };
+            Ok((result, end < last))
+        }
+    }
+
+    #[test]
+    fn package_search() {
+        let client = BuilderAPIClient::new("http://test.com", "", "", None).expect("valid client");
+
+        let sample_data = vec!["one_a", "one_b", "one_c", "one_d", "one_e", "two_a", "two_b",
+                               "two_c", "two_d", "two_e"];
+
+        let searcher = seach_generator(sample_data.as_slice(), 2);
+        let r = client.search_package_impl("one", 10, None, searcher)
+                      .expect("valid search");
+        assert_eq!(r.0.iter().map(|i| i.name.clone()).collect::<Vec<_>>(),
+                   vec!["one_a", "one_b", "one_c", "one_d", "one_e"]);
+        assert_eq!(r.1, 5);
+
+        let searcher = seach_generator(sample_data.as_slice(), 2);
+        let r = client.search_package_impl("_", 3, None, searcher)
+                      .expect("valid search");
+        assert_eq!(r.0.iter().map(|i| i.name.clone()).collect::<Vec<_>>(),
+                   vec!["one_a", "one_b", "one_c"]);
+        assert_eq!(r.1, 10);
+
+        let searcher = seach_generator(sample_data.as_slice(), 10);
+        let r = client.search_package_impl("a", 2, None, searcher)
+                      .expect("valid search");
+        assert_eq!(r.0.iter().map(|i| i.name.clone()).collect::<Vec<_>>(),
+                   vec!["one_a", "two_a"]);
+        assert_eq!(r.1, 2);
+
+        let searcher = seach_generator(sample_data.as_slice(), 10);
+        let r = client.search_package_impl("does_not_exist", 100, None, searcher)
+                      .expect("valid search");
+        assert_eq!(r.0.iter().map(|i| i.name.clone()).collect::<Vec<_>>(),
+                   Vec::<String>::new());
+        assert_eq!(r.1, 0);
+    }
+
+    #[ignore]
```

Comment:
Why is this ignored? I'm guessing it's because the test tends to take over 60s, and gives the `package_search_large has been running for over 60 seconds` warning, but on my machine it completes in ~90 seconds, so I think it's fine to put it in the normal suite. It doesn't ever fail, does it?

I'd say `#[ignored]` is for things that are inconsistent, but we'd still like to fix them up. Tests like that won't be noticed if they catch a regression, I'd want to know if this started failing.

Reply:
No, it does not every fail. This relates to our conversation on benchmarking. Ideally, this would be a benchmark and could be run separately from the tests. I like tests to have as fast a feedback loop as possible, but this is probably abusing the `ignore` attribute.